src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php

```diff
@@ -0,0 +1,312 @@
+    }
+
+    /**
+     * @When I name it :name
+     */
+    public function iNameItIn($name)
+    {
+        $this->createPage->nameIt($name);
+    }
+
+    /**
+     * @When I rename it to :name
+     */
+    public function iRenameItTo($name)
+    {
+        $this->createPage->nameIt($name);
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt()
+    {
+        $this->createPage->create();
+    }
+
+    /**
+     * @When /^I specify its price to ("(?:€|£|\$)[^"]+")$/
+     */
+    public function iSpecifyItsPriceTo($price)
+    {
+        $this->createPage->specifyPrice($price);
+    }
+
+    /**
+     * @Then the :productVariantName variant of the :product product should appear in the shop
+     */
+    public function theProductVariantShouldAppearInTheShop($productVariantName, ProductInterface $product)
+    {
+        $this->iWantToViewAllVariantsOfThisProduct($product);
+
+        Assert::true(
+            $this->indexPage->isSingleResourceOnPage(['presentation' => $productVariantName]),
+            sprintf('The product variant with name %s has not been found.', $productVariantName)
+        );
+    }
+
+    /**
+     * @When /^I want to view all variants of (this product)$/
+     */
+    public function iWantToViewAllVariantsOfThisProduct(ProductInterface $product)
+    {
+        $this->indexPage->open(['productId' => $product->getId()]);
+    }
+
+    /**
+     * @Then I should see :numberOfProductVariants variants in the list
+     * @Then I should see :numberOfProductVariants variant in the list
+     */
+    public function iShouldSeeProductsInTheList($numberOfProductVariants)
+    {
+        $foundRows = $this->indexPage->countItems();
+
+        Assert::same(
+            $numberOfProductVariants,
+            $foundRows,
+            '%s rows with product variants should appear on page, %s rows has been found'
+        );
+    }
+
+    /**
+     * @When /^I delete the ("[^"]+" variant of product "[^"]+")$/
+     * @When /^I try to delete the ("[^"]+" variant of product "[^"]+")$/
+     */
+    public function iDeleteTheVariantOfProduct(ProductVariantInterface $productVariant)
+    {
+        $this->iWantToViewAllVariantsOfThisProduct($productVariant->getObject());
+
+        $this->indexPage->deleteResourceOnPage(['code' => $productVariant->getCode()]);
+    }
+
+    /**
+     * @Then /^(this variant) should not exist in the product catalog$/
+     */
+    public function productVariantShouldNotExist(ProductVariantInterface $product)
+    {
+        $this->iWantToViewAllVariantsOfThisProduct($product->getObject());
+
+        Assert::false(
+            $this->indexPage->isSingleResourceOnPage(['presentation' => $product->getPresentation()]),
+            sprintf('Product variant with code %s exists but should not.', $product->getPresentation())
+        );
+    }
+
+    /**
+     * @Then /^I should be notified that this (?:variant|product) is in use and cannot be deleted$/
```

Comment:
Is it necessary to allow to use also "product" word? Nevertheless, its ``ManagingProductVariantsContext`` and there should be only steps required for managing variants.